### PR TITLE
Clarify exception handling

### LINKED ISSUE #1366: Document that sprinkling onComplete/onSuccess in all routes is not a good idea
In the onComplete/onSuccess docs we're very dry and only say how it works: http://doc.akka.io/docs/akka-http/10.0.9/scala/http/routing-dsl/directives/future-directives/onSuccess.html#onsuccess

This results in people sprinkling these directives into each route where they handle some future and doing the `{ case Failure(...) => complete(500, ...` manually instead of using Exception handling http://doc.akka.io/docs/akka-http/10.0.9/java/http/routing-dsl/exception-handling.html

We should improve docs of those directives to explain that most of the time exception handling is the right way to do this. We should also point out that we have the with circuit breaker on complete which is more useful than just completing

### PR DESCRIPTION
**Pull request by sebastianharko**
Fixes #1366

@ktoso @raboof Let me know if this looks ok. BTW, wondering whyJenkins PR validation appears to be stuck ? "Jenkins PR Validation — Waiting for status to be reported" (has been like this for days)

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/directives/future-directives/onComplete.md [hunk 1]
```diff
@@ -15,4 +15,12 @@ To complete with a successful result automatically and just handle the failure r
...
-@@snip [FutureDirectivesExamplesSpec.scala](../../../../../../../test/scala/docs/http/scaladsl/server/directives/FutureDirectivesExamplesSpec.scala) { #onComplete }
\ No newline at end of file
+@@snip [FutureDirectivesExamplesSpec.scala](../../../../../../../test/scala/docs/http/scaladsl/server/directives/FutureDirectivesExamplesSpec.scala) { #onComplete }
+
+@@@ note
+Using the `onComplete` directive means that you'll have to explicitly and manually handle failure cases. Doing this for every route in your app will result in a lot of boilerplate code. Most of the time you want to use the @ref[Exception Handling](../../exception-handling.md) mechanism instead.
+@@@
+
+@@@ note
+The identically named `onComplete` method of Scala's `Future` (from the standard library) does not work at all in this context since it's just a method that returns `Unit` - whereas Akka HTTP's `onComplete` is a `Directive` that creates a `Route`.
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/future-directives/onComplete.md]: yup exactly, thanks :)

File: docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md [hunk 2]
```diff
@@ -62,10 +62,12 @@ Note that `IllegalRequestException`s' stack traces are not logged, since instanc
 normally contain enough information to provide a useful error message.
 
 @@@ note
-Users are strongly encouraged not to rely on the `ExceptionHandler` as a means
-of handling 'normally exceptional' situations.
 
-Exceptions are known to have a negative performance impact for cases
+Users are strongly encouraged not to rely on using the `ExceptionHandler` as a means of handling errors. By errors, we mean things that are an expected part of normal operations: for example, issues discovered during input validation. The `ExceptionHandler` is meant to be a means of handling failures. See [Failure vs Error](https://www.reactivemanifesto.org/glossary#Failure) in the glossary of the [Reactive Manifesto](https://www.reactivemanifesto.org).
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md]: Yeah exactly, well explained :)

**Comment by raboof** [comment]
OK TO TEST

File: docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md (hunk 2 above)
Comment (raboof) [reply docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md]: Should we point to `onComplete` and the `RejectionHandler` here as alternatives?

Comment (sebastianharko) [reply docs/src/main/paradox/scala/http/routing-dsl/exception-handling.md]: @raboof yes but maybe next iteration (don't have time this week) :-)

File: docs/src/main/paradox/java/http/routing-dsl/directives/future-directives/onComplete.md [hunk 3]
```diff
@@ -12,4 +12,8 @@ To complete with a successful result automatically and just handle the failure r
 
 ## Example
 
-@@snip [FutureDirectivesExamplesTest.java](../../../../../../../test/java/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java) { #onComplete }
\ No newline at end of file
+@@snip [FutureDirectivesExamplesTest.java](../../../../../../../test/java/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java) { #onComplete }
```
[the remark this answers is not in the document]
Comment (ktoso) [reply docs/src/main/paradox/java/http/routing-dsl/directives/future-directives/onComplete.md]: This has to be fixed now, should be `$test$/test/java/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java) { #onComplete }` I guess right?

[the remark this answers is not in the document]
Comment (raboof) [reply docs/src/main/paradox/java/http/routing-dsl/directives/future-directives/onComplete.md]: `$test$/java/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java` jup

File: docs/src/main/paradox/scala/http/routing-dsl/directives/future-directives/onComplete.md [hunk 4]
```diff
@@ -15,4 +15,12 @@ To complete with a successful result automatically and just handle the failure r
 
 ## Example
 
-@@snip [FutureDirectivesExamplesSpec.scala](../../../../../../../test/scala/docs/http/scaladsl/server/directives/FutureDirectivesExamplesSpec.scala) { #onComplete }
\ No newline at end of file
+@@snip [FutureDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/FutureDirectivesExamplesSpec.scala) { #onComplete }
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/future-directives/onComplete.md]: I changed the paths @raboof - should be good to merge now I guess?

Comment (raboof) [reply docs/src/main/paradox/scala/http/routing-dsl/directives/future-directives/onComplete.md]: :+1: if jenkins agrees :)

**Comment by ktoso** [comment]
Thanks a ton @sebastianharko !

**Pull request merged**